app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,84 @@
+class TweetsController < ApplicationController
+  
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    if  @user
+      erb :'tweets/show_tweet'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = User.find(session[:id])
+      erb :'tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+
+  get '/tweets' do
+    if !logged_in?
+      redirect '/login'
+    else
+      @user = User.find(session[:id])
+      erb :home
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (not just that any user is logged in) before allowing them to edit?